ledger/db/src/tx_out_store.rs

```diff
@@ -108,7 +118,14 @@ impl TxOutStore {
             self.tx_out_index_by_hash,
             &tx_out.hash(),
             &u64_to_key_bytes(index),
-            WriteFlags::empty(),
+            WriteFlags::NO_OVERWRITE,
```

Comment:
NO_OVERWRITE ensures we do not find ourselves doing things we don't want to be doing. The `tx_out_by_*` should never have duplicate entries.